### PR TITLE
Hook for file name for PDF invoice

### LINKED ISSUE #24844: Invoice file name formatting
#### Is your feature request related to a problem?
There is possible to use the hook actionInvoiceNumberFormatted for invoice number formatting. I'm frustrating that this functionality doesn't exist also for pdf invoice file name when the invoice is downloaded. I'd like to use the same pdf file name as  the invoice number. E.g., if the invoice is FA21059874 the I'd like to have the file name FA21059874.pdf for this invoice.

#### Describe the solution you'd like
Add the new hook for invoice pdf file name formatting.

### PR DESCRIPTION
**Pull request by juraj1000**
| Questions         | Answers
| ----------------- | -------------------------------------------------------
| Branch?           | develop 
| Description?      | Create new hook for PDF invoice file name
| Type?             | improvement
| Category?         | BO
| BC breaks?        | no
| Deprecations?     | no
| Fixed ticket?     | Fixes #24844
| How to test?      | Create new implementation for the hook in a new customer module. Go to BO > Sell > Orders > Orders > Select order with an invoice > Display invoice > check if PDF file name is changed according to implementation for the hook
| Possible impacts? | No

---
This change is [Reviewable](https://reviewable.io/reviews/prestashop/prestashop/24941)

### CODE REVIEW

File: classes/pdf/HTMLTemplateInvoice.php
Comment (Progi1984) [inline classes/pdf/HTMLTemplateInvoice.php]:
```suggestion
        if (is_string($invoice_file_name) && !empty($invoice_file_name)) {
```

**Review by Progi1984 — COMMENTED** [review]
Good job

**Review by Progi1984 — DISMISSED** [review]

**Review by PierreRambaud — DISMISSED** [review]

**Comment by PierreRambaud** [comment]
I approfve the change but this PR needs to be rebased :)

**Comment by juraj1000** [comment]
Hello,
I changed the branch back to develop yesterday, but still I can see there is a label Waiting for rebase. Is there anything what is necessary to do from my side?

**Comment by matks** [comment]
> Hello,
> I changed the branch back to develop yesterday, but still I can see there is a label Waiting for rebase. Is there anything what is necessary to do from my side?

You're just too fast for us 😄  we have to remove the label manually

However I see there is git conflicts 😞

**Comment by juraj1000** [comment]
I corrected php files and now it seems that source code is now correct. But there are two error checks
- PHP / PHP CS Fixer (pull_request) 
- Pull Request Validator / Block Merge Commits (pull_request) 
I don't understand what is necessary to do. Please could you send me info how I could I correct it?

**Comment by PierreRambaud** [comment]
> I corrected php files and now it seems that source code is now correct. But there are two error checks
> 
> * PHP / PHP CS Fixer (pull_request)
> * Pull Request Validator / Block Merge Commits (pull_request)
>   I don't understand what is necessary to do. Please could you send me info how I could I correct it?

Everything about PHP cs fixer is here: https://github.com/PrestaShop/PrestaShop/pull/24941/checks?check_run_id=2830153136
It's necessary because we want all the codebase to be written with the same format.

About merge commits, it's because commit names should give an idea of the nature and context of the change that has been done. The more details, the better! The commit name should be as unique and recognizable as your commit itself. So we prefer git rebase instead of git merge.

**Comment by juraj1000** [comment]
I'm lost, I did my best but I don't know how to rebase my commits. I use GitHub Desktop. When I choose menu Branch > Rebase current branch the button Rebase is not allowed. What's wrong? How could I solve the errors form the checks?

**Review by PierreRambaud — DISMISSED** [review]

**Comment by PierreRambaud** [comment]
> I'm lost, I did my best but I don't know how to rebase my commits. I use GitHub Desktop. When I choose menu Branch > Rebase current branch the button Rebase is not allowed. What's wrong? How could I solve the errors form the checks?

I did it for you ;)

**Comment by juraj1000** [comment]
Thanks @PierreRambaud, I don't know what I did incorrectly, I hope the next time I'll avoid a similar issue

**Review by atomiix — DISMISSED** [review]

**Review by kpodemski — DISMISSED** [review]
Good job 👍 

Could you add information about this hook to devdocs? Here's an example: https://github.com/PrestaShop/docs/pull/848/files

**Comment by juraj1000** [comment]
I can update documentation for the new hook. But it is not clear to me, will it be delivered in PS 1.7.8.0 or 1.7.9.0 ? I need this information for updating file list-of-hooks.md

**Comment by kpodemski** [comment]
@juraj1000 develop is `1.7.9`

**Comment by juraj1000** [comment]
@kpodemski  Which file has to be updated ? Your example seems to be valid for 1.7.8.0 and I need 1.7.9.0.  I made changes in the branch develop and there is no path /src/content .... for the file list-of-hooks.md

**Comment by kpodemski** [comment]
@juraj1000 

documentation is available here:
https://github.com/PrestaShop/docs

**Review by Progi1984 — COMMENTED** [review]
Hi @juraj1000, you have done a great work, would you mind to rebase your pull request because we don't accept merge commits in PR? Thanks

Reference : https://anavarre.net/how-to-rebase-a-github-pull-request/

**Comment by juraj1000** [comment]
Hi @Progi1984, 
I did rebase with force push. I hope it is correct. But I can see 2 errors in checks > "PHP / PHP CS Fixer (pull_request)" and "Pull Request Validator / Block Merge Commits (pull_request)". But I don't know how to solve it. Could you explain to me what I should do and how I should correct it?

**Comment by Progi1984** [comment]
@juraj1000 No sorry. Your PR always contains some merge commits. Could you rebase your PR from upstream develop and remove merge commits ? Another solution : create a new PR ;)

**Comment by khouloudbelguith** [comment]
Hi @juraj1000,

This PR is not ready for QA.
Thanks to check the Conflicting files.

Thank you!

**Comment by kpodemski** [comment]
Hello @juraj1000 

Could you create a new PR? I know that rebasing is sometimes tricky. Feel free to create a new PR, and you'll probably have quick approvals as the code seems to be completely fine.

**Comment by PierreRambaud** [comment]
Hi @juraj1000 ,

Since we had no news from you for more than 30 days, I'll close this pull request. 
Feel free to recreate a new pull request and if possible with a dedicated branch, we will be able to help you in order to resolve conflict :) 

Thanks!

**Pull request closed without merging**
